core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -74,10 +74,10 @@ class SparkEnv (
   @volatile private[spark] var isStopped = false
   private val pythonWorkers = mutable.HashMap[(String, Map[String, String]), PythonWorkerFactory]()
 
-  // A general, soft-reference map for metadata needed during HadoopRDD split computation
-  // (e.g., HadoopFileRDD uses this to cache JobConfs and InputFormats).
+  // A general, weak-reference map for metadata needed during HadoopRDD split computation
+  // (e.g., HadoopRDD uses this to cache JobConfs).
   private[spark] val hadoopJobMetadata =
-    CacheBuilder.newBuilder().softValues().build[String, AnyRef]().asMap()
+    CacheBuilder.newBuilder().weakValues().build[String, AnyRef]().asMap()
```

Comment:
is it better to put a size limitation for this cache?

Reply:
Use limited size, soft-reference cache can reduce the number of YGC than weak-reference.
But what size should it be limited to?
In fact, the driver rarely has the opportunity to reuse the jobconf of the cache, and it makes sense to share the jobconf in the executor.

Reply:
Whether or not a limit is in place (which could also be a good back-stop to prevent a huge cache), this could be fine - I think the only risk is that weak references are quite readily reclaimed, so this risks losing most of the caching.